### ISSUE TITLE
Gracefully fallback to subprocess when Python wrappers not available/insufficient

### ISSUE THREAD

**Issue by sushain97** [apertium/apertium-python#50]
This should at least fix the Windows build so that we're not going backwards.

**Comment by sushain97** [comment]
We should also fallback if we don't support the specified flags/combination of them.

**Comment by sushain97** [comment]
@Vaydheesh why is the Windows build still not working? You mentioned something about `lrx-proc` but I didn't really catch it.

**Comment by Vaydheesh** [comment]
https://ci.appveyor.com/project/sushain97/apertium-python/builds/25755769/job/sx0oreu75t67vq9l#L210
`lrx-proc -m` seems to be expecting input from `stdin` on windows. I tried building `lrx-proc` on windows itself, and the last working commit was https://github.com/apertium/apertium-lex-tools/commit/4165215069a708a6a2182a3d9c9a3c3defed6dd1 

Should I log messages when calling processes upon falling back to subprcoess implementation?
This would help in finding the exact cause in future.

**Comment by sushain97** [comment]
> https://ci.appveyor.com/project/sushain97/apertium-python/builds/25755769/job/sx0oreu75t67vq9l#L210
> `lrx-proc -m` seems to be expecting input from `stdin` on windows.

Ah. This was working before, right?

> I tried building `lrx-proc` on windows itself, and the last working commit was [apertium/apertium-lex-tools@4165215](https://github.com/apertium/apertium-lex-tools/commit/4165215069a708a6a2182a3d9c9a3c3defed6dd1)

As in, it doesn't build after this commit or it hangs only after this commit?

> Should I log messages when calling processes upon falling back to subprcoess implementation?

I'm not sure what you would log but go ahead and try? You can also run it locally, yes? If not: https://www.appveyor.com/docs/how-to/rdp-to-build-worker/.

**Comment by Vaydheesh** [comment]
> I'm not sure what you would log but go ahead and try?

I want to log the calling of each subprocess if python wrappers aren't available, like https://ci.appveyor.com/project/sushain97/apertium-python/builds/25755769/job/sx0oreu75t67vq9l#L210, RN its only showing `test_en_spa (tests.TestTranslate) ... ` https://ci.appveyor.com/project/sushain97/apertium-python/branch/master/job/9of59fyjciehgltr#L195
This might come handy if something fails during future builds

**Comment by sushain97** [comment]
Sure, feel free to log at debug level. Can you talk to @unhammer about what might be wrong now? It seems like an error unrelated to swig entirely (you can confirm this on Windows?).

**Comment by Vaydheesh** [comment]
I'll re-compile this(just to be sure) and then discuss this tomorrow

**Comment by Vaydheesh** [comment]
Can we remove -m flag, when calling `lrx-proc` during subprocess implementation. This should pass the test case in appveyor, until a fix is pushed on `apertium-lex-tools`

**Comment by sushain97** [comment]
Uhh, I'm opposed to arbitrarily removing flags with semantic value. Presumably removing the `-m` flag will change how a pair actually behaves and therefore is not OK.

**Comment by xavivars** [comment]
There were discussions on actually removing the mode without `-m`, so default acted the same way. We probably should wrap up that conversation

**Comment by sushain97** [comment]
@xavivars makes sense. That being said, we can't just willy nilly remove the `-m` from the mode pipe, yes?

**Comment by Vaydheesh** [comment]
@sushain97 Should I close this issue if any changes aren't required

**Comment by sushain97** [comment]
Is there another issue tracking adding tests for this? If so, go ahead and close this one, yes.

**Comment by Vaydheesh** [comment]
The subprocess tests are tracked here https://github.com/apertium/apertium-python/issues/54
